### PR TITLE
Bug fix pos id of more than 16 characters not allowed

### PR DESCRIPTION
**Pull request by gopipatell**

### CODE REVIEW

**Review by mx51damon — COMMENTED** [review]
Hi Gopi, for this one, I thought it might be a bit tricky. After I read through the Spice validation for the POSID field, I noticed that we have a few cases not covered for this posId field validation. (I am sorry, initially I didn't cover them all ..)

As you may see from the Spice code [here](https://github.com/mx51/spice/blob/5e84dc1eb0f0c23e49c798b5dc06d7e2b35708cd/src/renderer/utils/pairing.js#L19-L26), which covered a few validation cases and Espresso POS didn't cover it so far.

I was thinking we might need to cover all the validation cases for this ticket. (I am a bit concerned QA will come back to us later if we merged the current PR) What do you think?

**Comment by gopipatell** [comment]
> Hi Gopi, for this one, I thought it might be a bit tricky. After I read through the Spice validation for the POSID field, I noticed that we have a few cases not covered for this posId field validation. (I am sorry, initially I didn't cover them all ..)
> 
> As you may see from the Spice code [here](https://github.com/mx51/spice/blob/5e84dc1eb0f0c23e49c798b5dc06d7e2b35708cd/src/renderer/utils/pairing.js#L19-L26), which covered a few validation cases and Espresso POS didn't cover it so far.
> 
> I was thinking we might need to cover all the validation cases for this ticket. (I am a bit concerned QA will come back to us later if we merged the current PR) What do you think?

As per the Jira ticket there was only one validation mentioned  so just did that validation. Happy to copy Spice validation if that is what everyone agrees should be there in posID validation. Also need to update jira ticket if we are doing that.

File: src/components/CustomTextField/index.tsx [hunk 1]
```diff
@@ -27,6 +28,8 @@ function CustomTextField({
       fullWidth={fullWidth}
       helperText={helperText}
       InputProps={InputProps}
+      // eslint-disable-next-line react/jsx-no-duplicate-props
```
Comment (mx51damon) [inline src/components/CustomTextField/index.tsx:31]: Hi Gopi, I was thinking shall we introduce this rule: `"react/jsx-no-duplicate-props": [1, { "ignoreCase": false }]` for eslint, so later if we have some other attributes like `otherAttributes` and `OtherAttributes`, the rule will also be able to catch it. The core reason is we don't need to put this line `// eslint-disable-next-line react/jsx-no-duplicate-props` everytime.

BTW, it's an optional change. 🤗

**Review by mx51damon — APPROVED** [review]
lgtm, just one question wish to get confirmed 🤗.

File: src/components/CustomTextField/index.tsx (hunk 1 above)
Comment (gopipatell) [reply src/components/CustomTextField/index.tsx:31]: Good idea will make that change

File: src/utils/validators/pairFormValidators.ts [hunk 2]
```diff
@@ -34,11 +34,16 @@ function serialNumberValidator(value: string): boolean {
   return valueWithoutDash.length > 0 && valueWithoutDash.length === 9;
 }
 
+function postIdValidator(value: string): boolean {
+  return !!value.match(numberCharactersRegex) && value.length <= 16;
```
Comment (mx51damon) [inline src/utils/validators/pairFormValidators.ts:38]: Option chaining here for preventing `undefined.length` case occur?

**Review by mx51damon — COMMENTED** [review]
Hi Gopi, after I checkout to your branch and test the code, I am still allowed to input the same posId value, as you can see from the screenshot here:
image

**Review by mx51damon — APPROVED** [review]
Nice work ~~

**Pull request merged**